src/shapes/Image.ts

```diff
@@ -602,7 +602,8 @@ export class FabricImage<
     ctx: CanvasRenderingContext2D
   ) {
     ctx.imageSmoothingEnabled = this.imageSmoothing;
-    // @ts-expect-error TS doesn't respect this type casting
+    // cant use ts-expect-error because of ts 5.3 cross check
+    // @ts-ignore TS doesn't respect this type casting
     super.drawCacheOnCanvas(ctx);
```

Comment:
This on the called method would fix the issue without the need of silencing the error:

```ts
drawCacheOnCanvas(ctx: CanvasRenderingContext2D) {
    const self = this as TCachedFabricObject;

    ctx.scale(1 / self.zoomX, 1 / self.zoomY);
    ctx.drawImage(
      self._cacheCanvas,
      -self.cacheTranslationX,
      -self.cacheTranslationY
    );
}
```

Reply:
It makes me mad that typescript has to error (not warn!!!!) on unused error directives.
So you can silence errors ( dangerous ) but you can't silence error silencers for good code.

Reply:
I'm not following you. If you use `ts-expect-error` then I like that it errors if there isn't actually a type error. What you're trying to do with typedoc while keeping the current TS version is not really a common use case for them.

Reply:
Yes i know is not common, but is still silly that i cant keep that as a warning and has to block the compilation.

Reply:
Dows typedoc allow to specify the TS instead? So that it's aligned with fabric

Reply:
maybe. But this is the shortest and more logical path.
No code changes, just annotation changes.
Not fiddling with dependencies of dependencies of astro, just make it run.